prism_compile.c

```
@@ -7811,6 +7811,14 @@ VALUE
 pm_parse_file(pm_parse_result_t *result, VALUE filepath)
 {
     if (!pm_string_mapped_init(&result->input, RSTRING_PTR(filepath))) {
+#ifdef _WIN32
+        int e = rb_w32_map_errno(GetLastError());
+#else
+        int e = errno;
+#endif
+
+        rb_syserr_fail(e, RSTRING_PTR(filepath));
```

We want to return an error from this function so that callers can do cleanup (free anything they allocated). Could you create the error from this function and return it instead?

---

Oh no, I didn't even remove the following line. OK I think this is fixed now.